x-pack/platform/plugins/private/upgrade_assistant/server/config.ts

```diff
@@ -21,6 +21,22 @@ const configSchema = schema.object({
     serverless: schema.boolean({ defaultValue: true }),
   }),
 
+  /**
+   * Exlcude certain data streams or backing indices from getting certain correctiveActions.
+   * The key is the data stream pattern and the value is an array of corrective actions to exclude.
+   *
+   * Exclude readOnly data streams from getting read-only corrective actions.
+   * This is needed to avoid breaking certain built-in/system functionality that might rely on writing to these data streams.
+   * Example (excludes read-only corrective actions for 7_17_data_stream):
+   * xpack.upgrade_assistant.dataStreamExclusions:
+   *    7_17_data_stream: ["readOnly"]
```

Comment:
With this configuration it is possible to provide something like:

```
'[.logs-deprecation.elasticsearch-default]': ["readOnly", "reindex"]
```

I tested locally and this has the effect of removing the deprecation. I suppose this is OK, just wanted to call out this functionality. Don't think we need to change it, this configuration should remain very much internal and possibly used as a backdoor to assist users.

Reply:
Yes this is intentional. I do filter out the data stream if both corrective actions are removed.